build-system/tasks/e2e/mocha-ci-reporter.js

```diff
@@ -27,10 +27,10 @@ const {Base} = mocha.reporters;
  */
 function ciReporter(runner, options) {
   Base.call(this, runner, options);
-  this._mochaDotsReporter = new MochaDotsReporter(runner, options);
-  this._jsonReporter = new JsonReporter(runner, options);
+  this._mochaDotsReporter = new MochaDotsReporter(runner);
+  this._jsonReporter = new JsonReporter(runner);
   this._mochaJunitReporter = new MochaJUnitReporter(runner, options);
-  return this;
+  return /** @type {*} */ (this);
```

Comment:
These types are actually not a complete union in the types (stats, failures, runner, and epilogue are all missing on ciReporter). I can add those properties something like this. Otherwise casting to `*` then returning with a return type of `MochaDotsReporter` means that any consumer of this will treat the returned value as `MochaDotsReporter`.
```
function ciReporter(runner, options) {
  this.runner = runner;
  Base.call(this, runner, options);
  this._mochaDotsReporter = new MochaDotsReporter(runner);
  this._jsonReporter = new JsonReporter(runner);
  this._mochaJunitReporter = new MochaJUnitReporter(runner, options);
  this.stats = this._mochaJunitReporter.stats;
  this.failures = this._mochaJunitReporter.failures;
  this.epilogue = () => {
    this._mochaDotsReporter.epilogue();
    this._jsonReporter.epilogue();
    this._mochaJunitReporter.epilogue();
  };
  return this;
}
```

I was a little hesitant to patch this incompatibility but if you view it as preferable to the cast I'll make the change.

Reply:
Leaving the cast as-is SGTM.